flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1088,9 +1152,14 @@ class DiscoverVolumesMethod(Names):
 
 @implementer(ILocalState)
 class FakeBlockDeviceDeployerLocalState(PClass):
+    """
+    This class is a terrible idea and should be thrown out ASAP to be
+    replaced by the real thing.
+    """
     node_state = field(type=NodeState, mandatory=True)
     nonmanifest_datasets = field(type=NonManifestDatasets, mandatory=True)
     volumes = pvector_field(BlockDeviceVolume)
+    discovered_datasets = pmap_field(UUID, DiscoveredDataset)
```

Comment:
Unfortunately the real implementation now doesn't have this member. I think it only has "datasets".